src/Symfony/Component/Console/CommandLoader/ContainerCommandLoader.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Symfony\Component\Console\CommandLoader;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * Loads commands from a PSR-11 container.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class ContainerCommandLoader implements CommandLoaderInterface
+{
+    private $container;
+    private $names;
+
+    public function __construct(ContainerInterface $container, array $names)
```

Comment:
Should `$names` be a map of (command name) => (service name) rather than forcing the command name to be the same of the service name ?

Or maybe I misunderstood what this class is doing?